automation/taskcluster/lib/tasks.py

```diff
@@ -129,11 +129,22 @@ def craft_assemble_raptor_task(self, variant):
             },
         )
 
-    def craft_assemble_task(self, variant):
+    def craft_assemble_pr_task(self, variant, run_coverage=False):
+        coverage_flag = '-Pcoverage ' if run_coverage else ''
+        assemble_gradle_command = ('{} assemble{}'.format(coverage_flag, variant.for_gradle_command),)
+        post_gradle_command = ('automation/taskcluster/upload_coverage_report.sh' if run_coverage else '',)
```

Comment:
Understanding the order of operations here is a little tough, but I don't want to block your PR - I can take a look when I update the `nimbledroid` secret